java/dataset/src/main/java/org/apache/arrow/dataset/source/Dataset.java

```diff
@@ -32,4 +32,14 @@ public interface Dataset extends AutoCloseable {
    * @return the Scanner instance
    */
   Scanner newScan(ScanOptions options);
+
+  /**
+   * Create a new Scanner, using the provided options,
+   * that contains the binary representation of the Substrait
+   * Extended Expression.
+   *
+   * @param options options used during creating Scanner
+   * @return the Scanner instance
+   */
+  Scanner newSubstraitScan(ScanOptions options);
```

Comment:
The point of having a ScanOptions object is that we shouldn't need a second function like this, right? newScan should just do the right thing based on the options.

Reply:
changed